File: app/src/main/java/com/example/domain/type/FizzBuzzType01.java
package com.example.domain.type;

import com.example.domain.model.FizzBuzzValue;

public class FizzBuzzType01 extends FizzBuzzType {
  public FizzBuzzValue generate(int number) {
    if (fizz(number) && buzz(number)) {
      return new FizzBuzzValue(number, "FizzBuzz");
    }
    if (fizz(number)) {
      return new FizzBuzzValue(number, "Fizz");
    }
    if (buzz(number)) {
      return new FizzBuzzValue(number, "Buzz");
    }
    return new FizzBuzzValue(number, String.valueOf(number));
  }
}

File: app/fizzbuzz-scala/src/main/scala/domain/types/FizzBuzzType01.scala
package domain.types

import domain.model.FizzBuzzValue

object FizzBuzzType01 extends FizzBuzzType {
  override def generate(number: Int): FizzBuzzValue = {
    if (isFizzBuzz(number)) {
      FizzBuzzValue(number, "FizzBuzz")
    } else if (isFizz(number)) {
      FizzBuzzValue(number, "Fizz")
    } else if (isBuzz(number)) {
      FizzBuzzValue(number, "Buzz")
    } else {
      FizzBuzzValue(number, number.toString)
    }
  }
}
